.github/workflows/test.yml

```diff
@@ -70,3 +79,60 @@ jobs:
 
       - name: Ensure no files were modified as a result of the build
         run: git update-index --refresh && git diff-index --quiet HEAD -- || git diff --exit-code
+
+  plugin_test:
+    name: plugin test
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        go-version:
+          - 1.22.x
+          - 1.23.x
+
+    env:
+      # COSIGN_REPO: ${{ github.event_name == 'workflow_dispatch' && inputs.cosign_repo  || 'sigstore/cosign'}}
+      # COSIGN_REF: ${{ github.event_name == 'workflow_dispatch' && inputs.cosign_ref  || 'main'}}
+      # TODO: remove my personal forks' references.
+      COSIGN_REPO: ramonpetgrave64/cosign
+      COSIGN_REF: ramonpetgrave64-kms-plugin-cli
+
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          path: sigstore
+
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Comment:
Why do we need Cosign for these tests? Could we instead have a test that imports cliplugin and calls all of the implemented functions, `CreateKey`, etc? This would be more lightweight and easy to iterate on.

Reply:
We could do that, but we would have to trigger that unit test with a special build flag, since the invoked of the unit test would first need to compile a real plugin binary first, probably our localkms implementation, since it also uses our Dispatch functions. And then the workflow could trigger this unit test with the special build flag. We also planned to make this build flag configurable, so plugin authors could run sigstore unit tests against their own prebuilt plugin programs. 

I like to know we’re still compatible with cosign, so I think we should do both.

Reply:
Added in the in `signer_program_test.go`, Makefile, `test.yml`, and `e2e_test.yml`. We won't need to run it on both workflows, but I'm not sure where is more appropriate. wdyt?

Reply:
e2e_test tests the cloud provider integrations. That seems like a good place to have this test as well. Putting these tests besides build tags SGTM.

I would omit Cosign testing here, as this is a general-purpose library that doesn't need to be aware of its dependents.

Reply:
Alright, removed.